LayoutTests/fast/canvas/webgl/resources/tex-image-and-sub-image-2d-with-video.js

```diff
@@ -74,10 +74,10 @@ function generateTest(pixelFormat, pixelType, prologue) {
         // the right color.
         debug("Checking lower left corner");
         wtu.checkCanvasRect(gl, 4, 4, 2, 2, bottomColor,
-                            "shouldBe " + bottomColor);
+                            "shouldBe " + bottomColor, 64);
```

Comment:
Slop of 64 (error range 128) is too big. I't just an indication  that something's off.

Reply:
I picked 64 because the expected pure green 0,255,0 (presumably in raw Composite NTSC) comes out as sRGB 63,251,0.
And it is what I also see when just looking at the mp4 video directly and taking a color sample, see attached screenshot.
![Screenshot 2024-05-23 at 16 47 42](https://github.com/WebKit/WebKit/assets/8499548/4ca7707f-d936-4b58-9104-38cfac890c78)
Other suggestions on how to fix/tweak this test?

Reply:
Sure, but again: with error range of 128 per value, a random garbage is equally probable to pass as it is to fail. That kind of slop cannot be had!

So preferably:
The test in question should exist in WebGL CTS already.
The test in question should be removed if it fails.
The test in WebGL CTS should be studied, understood and fixed, if there is something wrong with it. It should be analyzed exactly what is the exact problem, and then fixed.
In the analysis, it should be considered how other browsers are able to pass the test. Do they not implement the feature you are implementing, or do they interpret the source .mp4 colorspace somehow differently.
Analyzing whether the other browsers understand the colorspace, it maybe helpful to use them to look at the test.
Consider following test file:
- show the video frame in a video tag
- show the video frmae through webgl without your feature
- show the video frame through weblg with your feature

Then try to analyze what behavior is correct, and why. Then articulate that.

Reply:
> I suppose I could just hard-code these exact actual values as expected in the test, would you prefer that?
If you consider your task:
We don't support unpackColorSpace. Implement it.
Now test that does not use unpackColorSpace starts to fail.
So obviously there's a behavior difference in the work that is unrelated to unpackColorSpace.
I wouldn't want to just side-step this fact. The test caught something that currently seems like a bug in the new implementation. 

If the test is wrong, fix the test. If the implementation is wrong, fix the implementation.  Both should be explained :) 
And the test cannot be such that it accepts random values.

Reply:
If the WebGL CTS test cannot be used due to timeouts, then maybe we need to fix this test.
We should also investigate what can be done with the timeouts..

Reply:
If we had #28371, would it mean that this issue would go away? E.g. if UNPACK_COLORSPACE_CONVERSION_WEBGL cannot ever affect video elements, unpackColorspace implementation cannot possibly affect video elements?

Reply:
Thanks for all the tips.

I believe the new implementation is correct, because I'm getting the same sRGB values (for this red-green video) as seen in other macOS apps and also in Chrome -- unless everybody is equally wrong!?
So I'll focus on the tests tomorrow.

Right, UNPACK_COLORSPACE_CONVERSION_WEBGL as it's currently implemented in WK could theoretically make the video tests work by preventing the new conversions due to unpackColorSpace, but I believe UNPACK_COLORSPACE_CONVERSION_WEBGL is not used in these tests, hence the new results we get when applying the new unpackColorSpace conversions.
If we pushed #28371 to correct UNPACK_COLORSPACE_CONVERSION_WEBGL to only work with images as per specs, then yeah unpackColorSpace would always affect video elements, so we'll definitely want these tests to correctly expect conversions.